splito/_split.py

```diff
@@ -0,0 +1,101 @@
+from enum import Enum, unique
+from typing import Optional, Sequence, Union
+
+import datamol as dm
+import numpy as np
+from sklearn.model_selection import ShuffleSplit
+
+from ._distribution_split import StratifiedDistributionSplit
+from ._kmeans_split import KMeansSplit
+from ._max_dissimilarity_split import MaxDissimilaritySplit
+from ._min_max_split import MolecularMinMaxSplit
+from ._molecular_weight import MolecularWeightSplit
+from ._perimeter_split import PerimeterSplit
+from ._scaffold_split import ScaffoldSplit
+
+
+@unique
+class SimpleSplittingMethod(Enum):
+    RANDOM = ShuffleSplit
+    KMEANS = KMeansSplit
+    PERIMETER = PerimeterSplit
+    MAX_DISSIMILARITY = MaxDissimilaritySplit
+    SCAFFOLD = ScaffoldSplit
+    STRATIFIED_DISTRIBUTION = StratifiedDistributionSplit
+    MOLECULAR_WEIGHT = MolecularWeightSplit
+    MIN_MAX_DIVERSITY_SPLIT = MolecularMinMaxSplit
+
+
+def train_test_split(
+    X: np.ndarray,
+    y: np.ndarray,
+    molecules: Optional[Sequence[Union[str, dm.Mol]]] = None,
```

Comment:
I understand here the goal is to mimic scikit learn, but I don't really understand what is X in that context? It seems to only support arrays, i.e. if the molecules are represented as vectors.

What if someone wants to use PyG objects? Or what if someone just wants to split a set of smiles to create fixed CSV's? Then this function doesn't work and providing `X` is ambiguous.

It would be ideal if you can just provide the `train_indices` and `test_indices` and let the person split afterwards.

Of course, you could say, "If the user wants indices, they can provide `X` as indices", but it's not straightforward and confusing.

Reply:
An alternative is to have another function `train_test_split_indices` that is called by this one.

Reply:
I like your second suggestion!